packages/beacon-node/src/chain/archiveStore/differentialState/execute.ts

```diff
@@ -13,10 +18,75 @@ export async function regenerateState(
   ctx.logger?.verbose("Regenerating state via state differential", {
     slot: target,
   });
+  const regenTimer = ctx.metrics?.regenTime.startTimer();
+  ctx.metrics?.regenRequestCount.inc();
 
-  const plan = buildStateRegenPlan(ctx.layers, target);
-  const artifacts = await fetchStateRegenArtifacts(ctx.db, plan, opts);
-  const finalState = await applyStateRegenPlan(ctx, plan, artifacts);
+  try {
+    const plan = buildStateRegenPlan(ctx.layers, target);
+    const artifacts = await fetchStateRegenArtifacts({db: ctx.db, metrics: ctx.metrics}, plan, opts);
+    const finalState = await applyStateRegenPlan(ctx, plan, artifacts);
+    const state = snapshotToBeaconState(ctx, finalState);
 
-  return snapshotToBeaconState(ctx, finalState);
+    ctx.metrics?.regenSuccessCount.inc();
+    regenTimer?.();
+
+    return state;
+  } catch (err) {
+    ctx.metrics?.regenErrorCount.inc({reason: RegenErrorType.loadState});
+    throw err;
+  }
+}
+
+export async function storeDifferentialState(
```

Comment:
Where is this function used? If it is not related to metrics, please add it in another PR.